external-crates/move/crates/move-compiler/src/hlir/translate.rs

```diff
@@ -2201,8 +2202,18 @@ fn needs_freeze(context: &Context, sp!(_, actual): &H::Type, sp!(_, expected): &
                 Freeze::NotNeeded
             }
         }
+        (T::Single(sp!(_, H::SingleType_::Base(sp!(_, H::BaseType_::Unreachable)))), _expected) => {
+            Freeze::NotNeeded
+        }
         (_actual, _expected) => {
-            assert!(context.env.has_errors());
+            if !context.env.has_errors() {
+                println!("typing produced the following actual and expected types: ");
+                print!("actual: ");
+                _actual.print();
+                print!("expected: ");
+                _expected.print();
+                panic!("ICE HLIR freezing went wrong")
+            }
```

Comment:
Ah ah ah. This makes sense. Should we also check for `H::BaseType_::UnresolvedError` above?

Reply:
I am going to leave this for now, because my plan is to sweep through all the ICEs after the macro PR lands and have each actually leave enough useful information on it that we can ask for bug reports using the information.